src/test/java/net/snowflake/client/jdbc/MaxLobSizeLatestIT.java

```diff
@@ -0,0 +1,253 @@
+      for (int j = 0; j < resultFormats.length; j++) {
+        ret.add(new Object[] {lobSizes[i], resultFormats[j]});
+      }
+    }
+    return ret;
+  }
+
+  private final int lobSize;
+
+  private final String resultFormat;
+
+  public MaxLobSizeLatestIT(int lobSize, String resultFormat) {
+    this.lobSize = lobSize;
+    this.resultFormat = resultFormat;
+  }
+
+  private static String tableName = "my_lob_test";
+  private static String enableMaxLobSize =
+      "alter session set FEATURE_INCREASED_MAX_LOB_SIZE_IN_MEMORY = 'ENABLED'";
+  private static String executeInsert = "insert into " + tableName + " (c1, c2, c3) values (";
+  private static String executePreparedStatementInsert = executeInsert + "?, ?, ?)";
+  private static String selectQuery = "select * from " + tableName;
+
+  private static String generateRandomString(int stringSize) {
+    RandomStringGenerator randomStringGenerator =
+        new RandomStringGenerator.Builder().withinRange('a', 'z').build();
+    return randomStringGenerator.generate(stringSize);
+  }
+
+  private static void setResultFormat(Connection con, String format) throws SQLException {
+    try (Statement stmt = con.createStatement()) {
+      stmt.execute("alter session set jdbc_query_result_format = '" + format + "'");
+    }
+  }
+
+  private void createTable(int lobSize, Connection con) throws SQLException {
+    try (Statement stmt = con.createStatement()) {
+      String createTableQuery =
+          "create or replace table "
+              + tableName
+              + " (c1 varchar, c2 varchar("
+              + lobSize
+              + "), c3 int)";
+      stmt.execute(createTableQuery);
+    }
+  }
+
+  private void insertQuery(String varCharValue, int intValue, Connection con) throws SQLException {
+    try (Statement stmt = con.createStatement()) {
+      stmt.executeUpdate(
+          executeInsert + "'" + varCharValue + "', '" + varCharValue + "', " + intValue + ")");
+    }
+  }
+
+  private void preparedInsertQuery(String varCharValue, int intValue, Connection con)
+      throws SQLException {
+    try (PreparedStatement pstmt = con.prepareStatement(executePreparedStatementInsert)) {
+      pstmt.setString(1, varCharValue);
+      pstmt.setString(2, varCharValue);
+      pstmt.setInt(3, intValue);
+
+      pstmt.execute();
+    }
+  }
+
+  @After
+  public void tearDown() throws SQLException {
+    try (Connection con = BaseJDBCTest.getConnection();
+        Statement stmt = con.createStatement()) {
+      stmt.execute("Drop table if exists " + tableName);
+    }
+  }
+
+  @Test
+  public void testStandardInsertAndSelectWithMaxLobSizeEnabled() throws SQLException {
+    try (Connection con = BaseJDBCTest.getConnection();
+        Statement stmt = con.createStatement()) {
+      // TODO: Uncomment when ready to test with max lob size enabled.
+      // stmt.execute(enableMaxLobSize);
+
+      setResultFormat(con, resultFormat);
+      createTable(lobSize, con);
```

Comment:
let's create table once in constructor (then for each parameters set it will be recreated)
then drop table can be moved to `@AfterClass`